File: 项目文件/backend/app/schemas/project_event.py
"""项目事件 Pydantic 模型"""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class ProjectEventCreate(BaseModel):
    """创建项目事件请求"""

    project_id: uuid.UUID
    number: str
    event_type: str
    title: str
    details: dict[str, Any] = {}


class ProjectEventUpdate(BaseModel):
    """更新项目事件请求"""

    number: str | None = None
    event_type: str | None = None
    title: str | None = None
    details: dict[str, Any] | None = None


class ProjectEventResponse(BaseModel):
    """项目事件响应"""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    number: str
    event_type: str
    title: str
    details: dict[str, Any]
    operator_id: uuid.UUID
    created_at: datetime


class ProjectEventListResponse(BaseModel):
    """项目事件列表响应"""

    items: list[ProjectEventResponse]
    total: int

File: 项目文件/frontend/src/types/project-event.ts
/** 项目事件类型定义 */

export interface ProjectEvent {
  id: string;
  project_id: string;
  number: string;
  event_type: string;
  title: string;
  details: Record<string, unknown>;
  operator_id: string;
  created_at: string;
}

export interface ProjectEventCreate {
  project_id: string;
  number: string;
  event_type: string;
  title: string;
  details?: Record<string, unknown>;
}

export interface ProjectEventUpdate {
  number?: string;
  event_type?: string;
  title?: string;
  details?: Record<string, unknown>;
}

export interface ProjectEventListResponse {
  items: ProjectEvent[];
  total: number;
}
